ArrtSource/Storage/FileUploader.cpp

```diff
@@ -58,21 +79,133 @@ void FileUploader::UploadFilesAsync(const QDir& sourceRootDirectory, const QStri
     std::thread(std::bind(asyncCallback, batchStart, sourceFilePaths.count())).detach();
 }
 
+void FileUploader::NotifyBytesRead(int64_t bytes)
+{
```

Comment:
out of interest, how often is this function called?

Reply:
Every 4 MB.